apptail/storage/tail_tracker.go

```diff
@@ -0,0 +1,258 @@
+package storage
+
+import (
+	"fmt"
+	"github.com/ActiveState/log"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+type Tracker interface {
+	Update(instKey string, childKey string, childVal int64)
+	LoadTailers()
+	Remove(key string)
+	RegisterInstance(instKey string)
+	InitializeChildNode(instKey string, childkey string, offSet int64)
+	Commit() error
+	StartSubmissionTimer(persistInterval time.Duration)
+	IsInstanceRegistered(instKey string) bool
+	IsChildNodeInitialized(instKey string, childkey string) bool
+	GetFileCachedOffset(instkey string, fname string) int64
+	getBuffer() ([]byte, error)
+	CleanUp(clenups map[string]bool)
+}
+
+type BoxedInt64 struct{ V int64 }
+
+type TailNode map[string]*BoxedInt64
+
+type Tailer struct {
+	Instances map[string]TailNode
+}
+
+type tracker struct {
+	storage       Storage
+	Cached        *Tailer // do not expose this, it should ONLY be updated via Tracker methods
+	mux           *sync.Mutex
+	timerStopChan chan struct{} // used to send quit signal to time
+}
+
+var (
+	MinIOTicker = 5 * time.Second
+)
+
+func NewTracker(s Storage) Tracker {
+
+	return &tracker{
+		storage: s,
+		mux:     &sync.Mutex{},
+		Cached: &Tailer{
```

Comment:
Uppercase => public
Lowercase => private 
 although there is a commnet that says "// do not expose this, it should ONLY be updated via Tracker methods", It's still not accesible, I mean not to put it in an interface.